### ISSUE TITLE
Median Frequency Balancing for CE loss

### ISSUE THREAD

**Issue by MABatin** [open-mmlab/mmsegmentation#2271]
Hi, I wanted to implement median frequency balancing for CE loss where the loss of each class is weighted by the ratio of median class frequency and actual frequency of the class. But I am at a loss (pun intended) as to which files to modify to implement such technique for Cross Entropy Loss. In particular, how do I get the frequency of each class at run-time, or do I compute it beforehand and pass it as a .npy argument for CrossEntropy class?

Any suggestion/help would be much appreciated.

**Comment by MengzhangLI** [comment]
Currently we do not provide tools to calculate frequency of each class in dataset so you have to calculate on your own.

If you get the median frequency, you can set weight for ce loss, here is [doc](https://github.com/open-mmlab/mmsegmentation/blob/master/docs/en/tutorials/training_tricks.md#class-balanced-loss).

**Comment by MABatin** [comment]
@MengzhangLI thanks for the reply. On a side note, any chance of implementing Unet++ model in this repo anytime soon?

**Comment by MengzhangLI** [comment]
> 

We do not plan to support UNet++ in the next few month because we are supporting some SOTA models in semantic segmentation area like MaskFormer, Mask2Former and SegNeXt. But if you create a PR of UNet++, we would response you quickly because community pr is high priority for us.

If you don't mind, welcome make a pr for anything you want to add in MMSegmentation.

Best,

**Comment by MABatin** [comment]
Many thanks for the information.
